cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```
@@ -136,6 +136,7 @@ public void create(HttpRequest request, final HttpResponder responder,
     // Enable ad-hoc exploration of dataset
     // Note: today explore enable is not transactional with dataset create - CDAP-8
     try {
+      datasetExploreFacade.disableExplore(name);
```

So it just fails silently if there was no Hive table for ``name``?